### ISSUE TITLE
Could this plugin support autocomplete in .js or .ts files?

### ISSUE THREAD

**Issue by ticktockreed** [vunguyentuan/vscode-css-variables#4]
Thanks for the plugin, works great for CSS and preprocessor CSS. 

It would be great to have access to CSS vars via autocomplete within .js and .ts files. 

How feasible is this?

**Comment by phoenisx** [comment]
How do u intend to use css variables in JS and TS??

**Comment by ticktockreed** [comment]
We're using EmotionJS in some projects and Aphrodite in others. So would need access within .js or .ts files.

See example below:

```
// aphrodite example - moduleStyle.js 

import { StyleSheet } from 'aphrodite/no-important';
export default StyleSheet.create({
  title: {
    color: var(--dp_text)
  },
});

```

```
emotionJS example -  module.style.ts

const primaryButton = (theme: TLTheme) => css`
  background-color: var(--dp_button_primary_background);
  color: var(--dp_text_inverted);
  :hover {
    background-color: var(--dp_button_primary_background_hover);
  }

  :active {
    background-color: var(--dp_button_primary_background_pressed);
  }
`;
```

**Comment by phoenisx** [comment]
I understand your concern!

I am building my own extension to support CSS Variables and CSS Module IntelliSense which you can find [HERE](https://github.com/willofindie/vscode-cssvar). It's still in the alpha stage. It works great with plain CSS and other CSS language extensions like SCSS etc.

I tried your scenario in my extension, it works but with a hack. I see VSCode API fails to trigger autocomplete if JS file is opened. I mean, unless a CSS file is opened once the Autocomplete doesn't trigger at all.

Check the below illustration, the IntelliSense triggers but only after you open a CSS or SCSS or a similar file.
![issue](https://user-images.githubusercontent.com/11786283/113185668-5aabc400-9274-11eb-80db-73943fd2278b.gif)

I will add this to my TODO list. I was thinking this would be a simple task, but it looks like it needs re-thinking for my extension and also needs re-architecting my whole extension.

You can open an issue (if u want to) [HERE](https://github.com/willofindie/vscode-cssvar/issues)

**Comment by phoenisx** [comment]
![js](https://user-images.githubusercontent.com/11786283/113474063-1ba38b80-948b-11eb-8de3-eddb9024b011.gif)

I hope this is what you were looking for. I fixed it in my extension 😅 🎉 
Thanks for making me aware of this feature 👍🏾

**Comment by vunguyentuan** [comment]
> Thanks for the plugin, works great for CSS and preprocessor CSS.
> 
> It would be great to have access to CSS vars via autocomplete within .js and .ts files.
> 
> How feasible is this?

Yes it is feasible but it will very annoying to other people because they dont want to get css suggestion on their ts, or js files.

I will try to allow users to decide that files they want to get autocomplete :)

**Comment by vunguyentuan** [comment]
@ticktockreed this should work now, thanks for the feedback.
The new version now supports goto definition and simplifies config or no config.

![screencast 2021-12-22 12-39-21](https://user-images.githubusercontent.com/7245986/147036726-1f272e8c-8a31-41d3-a27d-fb332f2da718.gif)

**Comment by PixeledCode** [comment]
> @ticktockreed this should work now, thanks for the feedback. The new version now supports goto definition and simplifies config or no config.
> 
> ![screencast 2021-12-22 12-39-21](https://user-images.githubusercontent.com/7245986/147036726-1f272e8c-8a31-41d3-a27d-fb332f2da718.gif) [ ![screencast 2021-12-22 12-39-21](https://user-images.githubusercontent.com/7245986/147036726-1f272e8c-8a31-41d3-a27d-fb332f2da718.gif) ](https://user-images.githubusercontent.com/7245986/147036726-1f272e8c-8a31-41d3-a27d-fb332f2da718.gif) [ ](https://user-images.githubusercontent.com/7245986/147036726-1f272e8c-8a31-41d3-a27d-fb332f2da718.gif)

Sorry for responding later after the issue is closed, but I cannot make it work with styled components. I created a `.css` file in the repo, which I tried by both methods, including and not including in `_app` file but I am not able to start any auto-completion

**Comment by karlhorky** [comment]
@vunguyentuan would you consider allowing disabling this feature for projects that do not use CSS-in-JS?

For example, adding a new setting called `cssVariables.autocompleteInLanguages`, default values below:

```json
{
  "cssVariables.autocompleteInLanguages": [
    "css",
    "less",
    "scss",
    "postcss",
    "javascript",
    "javascriptreact",
    "typescriptreact"
  ],
}
```

Or, if this is based on file extension instead of language in the file, a new setting called `cssVariables.autocompleteInFiles`, default values:

```json
{
  "cssVariables.autocompleteInFiles": [
    "*.css",
    "*.scss",
    "*.sass",
    "*.less",
    "*.js",
    "*.jsx",
    "*.ts",
    "*.tsx"
  ]
}
```

**Comment by karlhorky** [comment]
I can also create a new issue if you would prefer.

**Comment by karlhorky** [comment]
Opened #58 to track this.
